### PR TITLE
Update versions and artifact ids for the file spring-reference.yml

### PR DESCRIPTION
**Pull request by moarychan**
Update the version preview for the SCA 6.0 milestone version and add new artifact ids for Azure Database support.

Note:
- The `azure-toolkit-for-intellij` can not support multiple major versions for the same artifact id, but the `versionPreview` can be used for a new version, which can be a new major GA version or a milestone version.
 - `springProperties.**` will not be read for the toolkit, this is just a note for us to record the version support.
 - Add below new artifacts:
     - spring-cloud-azure-starter-keyvault
     - spring-cloud-azure-starter-storage
     - spring-cloud-azure-starter-jdbc-mysql
     - spring-cloud-azure-starter-jdbc-postgresql

### CODE REVIEW

**Comment by saragluna** [comment]
@moarychan the previous version of this file is generated using this tool https://github.com/saragluna/spring-reference-intellij. We should update that project's metadata as well? Otherwise, we will lose these changes the next time we use that tool.

**Comment by saragluna** [comment]
But what's the point for us to add 6.x to this file, since it won't be able to support two major versions.

**Comment by moarychan** [comment]
> But what's the point for us to add 6.x to this file, since it won't be able to support two major versions.

Since this tool will not process the compatibility with multi-major Spring Boot versions, the `versionPreview` of each artifact is just an extra option for version selection, currently, we can only use this property to guide developers to know what new versions can be used.

File: sdk/spring/spring-reference.yml [hunk 1]
```diff
@@ -186,17 +202,14 @@
...
           description: |-
             Microsoft's Spring Boot Starter helps developers to finish the auto-configuration of
             Spring Data Azure Cosmos DB, which enables developers to easily integrate with Azure Cosmos
             DB SQL API using Spring Data.
           type: spring
           links:
-            github: https://github.com/Azure/azure-sdk-for-java/tree/spring-cloud-azure_4.4.1/sdk/spring/spring-cloud-azure-starter-data-cosmos
             msdocs: https://docs.microsoft.com/azure/developer/java/spring-framework/configure-spring-boot-starter-java-app-with-cosmos-db
             repopath: https://search.maven.org/artifact/com.azure.spring/spring-cloud-azure-starter-data-cosmos
-            sample: https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.4.1/cosmos/spring-cloud-azure-starter-data-cosmos/
```
Comment (saragluna) [inline sdk/spring/spring-reference.yml:199]: Why remove this?

File: sdk/spring/spring-reference.yml [hunk 2]
```diff
@@ -291,17 +348,19 @@
...
           groupId: com.azure
           versionGA: 2.7.0
-          versionPreview: 1.0.0-beta.7
           type: client
           links:
             javadoc: https://azuresdkdocs.blob.core.windows.net/$web/java/azure-security-keyvault-jca/2.7.0/index.html
             github: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/keyvault/azure-security-keyvault-jca
             msdocs: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/keyvault/azure-security-keyvault-jca/README.md
             repopath: https://search.maven.org/artifact/com.azure/azure-security-keyvault-jca
-            sample: https://github.com/Azure-Samples/azure-spring-boot-samples/tree/main/keyvault/azure-securtiy-keyvault-jca
```
Comment (saragluna) [inline sdk/spring/spring-reference.yml:304]: Why remove this?

**Pull request merged**